iothub/device/tests/Authentication/IotHubConnectionStringBuilderTests.cs

```diff
@@ -214,38 +214,30 @@ public void IotHubConnectionStringBuilder_ParamConnectionString_ParsesGatewayHos
         public void IotHubConnectionStringBuilder_ParamConnectionString_MissingHostName_Throws()
         {
             var connectionString = $"DeviceId={DeviceId};SharedAccessKey={SharedAccessKey}";
-            var csBuilder = IotHubConnectionStringBuilder.Create(connectionString);
+            var csBuilder = new IotHubConnectionStringBuilder(connectionString);
         }
 
         [TestMethod]
-        [ExpectedException(typeof(ArgumentException))]
+        [ExpectedException(typeof(ArgumentNullException))]
         public void IotHubConnectionStringBuilder_ParamConnectionString_MissingDeviceId_Throws()
         {
             var connectionString = $"HostName={HostName};SharedAccessKey={SharedAccessKey}";
-            var csBuilder = IotHubConnectionStringBuilder.Create(connectionString);
-        }
-
-        [TestMethod]
-        [ExpectedException(typeof(ArgumentException))]
-        public void IotHubConnectionStringBuilder_ParamConnectionString_IncludesBothSharedAcccess_Throws()
```

Comment:
Yes, this is an invalid connection string, but once initialized by the builder when an auth method is created, it nulls out the shared access key, making it valid. I find it hard to get very excited about this condition being caught when the code fixes it.

Thoughts?

Reply:
Looks like this is a set of testing different way of the invalid connection string. Make sense when this is initially created to make sure the API handles them. Maybe we can remove them for v2.

Reply:
Passing in both a shared access signature and shared access key, you'd expect the more specific credential (sas) to be used. That being the behavior exhibited by our SDK, I don't mind us not throwing an exception. It would be a good bonus if we could call this out somewhere (DeviceClient connection string constructor?).